### PR TITLE
Unit test using input and output proto files

### PR DESCRIPTION
**Pull request by rghetia**

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Also, this is super nice :)

File: testdata/inMetrics_ExportMetricsOfAllTypes.txt [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+  >
+  points: <
+    timestamp: <
+      seconds: 1543160298
+      nanos: 100000997
+    >
+    double_value: 35.5
+  >
+>
+---
```
Comment (songy23) [inline testdata/inMetrics_ExportMetricsOfAllTypes.txt]: Do you need these separators? I think you can use a metric list instead, something like
metric: <
  metric_descriptor: <...>
  timeseries: <...>
>
metric: <
  metric_descriptor: <...>
  timeseries: <...>
>
...

Then in test.go:
```go
var metrics []*metricspb.metric
proto.UnmarshallTxt(metrics, file)
```

Comment (songy23) [reply testdata/inMetrics_ExportMetricsOfAllTypes.txt]: Same for metric descriptor requests and time series requests.

**Comment by rghetia** [comment]
> Hope you will not get mad on me :), instead of having the "inMetrics" "outTS" as prefix can we have them as suffix?

No problem.

> Personally I prefer more the directory structure because then all files related to that test are in one directory.

I like the directory approach as well.

File: testdata/inMetrics_ExportMetricsOfAllTypes.txt (hunk 1 above)
Comment (rghetia) [reply testdata/inMetrics_ExportMetricsOfAllTypes.txt]: UnmarshalText only accepts Message
```go
func UnmarshalText(s string, pb Message) error {
```
I would have to create wrapper proto like
```
message MetricList {
  repeated opencensus.proto.metric.v1.Metric metric = 1;
}
```
Let me know if I missed something.

Comment (songy23) [reply testdata/inMetrics_ExportMetricsOfAllTypes.txt]: Got it - I didn't know that, LGTM then.

**Comment by rghetia** [comment]
@bogdandrutu PTAL.

**Pull request merged**